### PR TITLE
Refactor cut/copy functionality

### PR DESCRIPTION
**Pull request by max-winderbaum**
to reduce the responsibility of external handlers to just translating a
blockMap to text and html

### CODE REVIEW

File: src/component/handlers/edit/editOnCopy.js [hunk 1]
```diff
@@ -31,9 +31,29 @@ function editOnCopy(editor: DraftEditor, e: SyntheticClipboardEvent): void {
     return;
   }
 
-  editor.setClipboard(getFragmentFromSelection(editor._latestEditorState));
-  if (editor.props.onCopy) {
-    editor.props.onCopy(e);
+  const fragment = getFragmentFromSelection(editor._latestEditorState);
+  editor.setClipboard(fragment);
+  if (editor.props.convertBlockMapToClipboard) {
+    const clipboardDataToPush = editor.props.convertBlockMapToClipboard(fragment);
```
Comment (colinjeanne) [inline src/component/handlers/edit/editOnCopy.js]: We should move this into its own file and include it in editOnCopy and editOnCut.

**Review by colinjeanne — CHANGES_REQUESTED** [review]

File: src/component/handlers/edit/editOnCopy.js (hunk 1 above)
Comment (max-winderbaum) [reply src/component/handlers/edit/editOnCopy.js]: Oh wow yeah it is the same code, ty!

**Comment by intentionally-left-nil** [comment]
LGTM. This is a straight refactor, correct?

**Review by intentionally-left-nil — APPROVED** [review]

**Comment by max-winderbaum** [comment]
@AnilRedshift yes one step at a time :)

File: src/component/handlers/edit/setClipboardData.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  try {
+    clipboard.setData(`text/html`, html);
+    clipboard.setData(`text/plain`, text);
+  }
+  catch (err) {
+    clipboard.setData(`Text`, text);
+  }
+}
+
+module.exports = setClipboardData;
```
Comment (colinjeanne) [inline src/component/handlers/edit/setClipboardData.js]: Minor: looks like all other files end with an empty line, except this one.

**Review by colinjeanne — APPROVED** [review]
Minor comment, :ship:

File: src/component/handlers/edit/setClipboardData.js (hunk 2 above)
Comment (max-winderbaum) [reply src/component/handlers/edit/setClipboardData.js]: I'll fix and :ship:

**Pull request merged**